core/src/replay_stage.rs

```diff
@@ -721,11 +721,12 @@ impl ReplayStage {
 
         if let Some(ref voting_keypair) = voting_keypair {
             let node_keypair = cluster_info.read().unwrap().keypair.clone();
+            let authorized_voter_pubkey = voting_keypair.lock().unwrap().pubkey();
```

Comment:
Hard for me to see the context here - this function is huge! But concerned about that `lock()` right next to some code that's using a `RwLock` and taking the read lock. Will this change cause thread contention? Should it be using `RwLock` as well?